test/docfx.Test/config/ConfigTest.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json.Linq;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.Docs.Build
+{
+    public static class ConfigTest
+    {
+        [Fact]
+        public static void TestConigWithGlobalMetadata()
```

Comment:
Can these test be written as yaml specs? You can implement the basic metadata in build pipeline that merges file and global metadata and output them to a json file.

Reply:
want to keep here to avoid too large PR. the spec is also prone to change for now as they are not final output.